src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -176,15 +210,20 @@ public RTPEncodingDesc(
      * @param rtxSSRC The RTX SSRC for this layering/encoding.
      * @param tid temporal layer ID for this layering/encoding.
      * @param sid spatial layer ID for this layering/encoding.
+     * @param resolution the resolution of this encoding
      * @param dependencyEncodings  The {@link RTPEncodingDesc} on which this
      * layer depends.
      */
     public RTPEncodingDesc(
         MediaStreamTrackDesc track, int idx,
         long primarySSRC, long rtxSSRC,
         int tid, int sid,
+        int resolution,
         RTPEncodingDesc[] dependencyEncodings)
     {
+        // XXX we should be able to snif the actual resolution from the RTP
+        // packets.
```

Comment:
We could do this by inspecting the payload (of keyframes). This wouldn't work with PERC tho, in which case we should get it from signaling.